world.BlockRuntimeID panics when placing down Furnace

```
INFO[0000] Server running on [::]:4444.                 
coEthaniccc has joined the game
DEBU[0129] unhandled packet *packet.SetPlayerInventoryOptions{LeftInventoryTab:10 RightInventoryTab:4 Filtering:false InventoryLayout:6 CraftingLayout:0} from xxx:53436 
panic: cannot find block by non-0 hash of block block.Furnace{solid:block.solid{}, bassDrum:block.bassDrum{}, smelter:(*block.smelter)(0xc00148a540), Facing:0, Lit:false} [recovered]
        panic: cannot find block by non-0 hash of block block.Furnace{solid:block.solid{}, bassDrum:block.bassDrum{}, smelter:(*block.smelter)(0xc00148a540), Facing:0, Lit:false}

goroutine 56 [running]:
github.com/df-mc/dragonfly/server/session.(*Session).handlePackets.func1()
        /root/dragonfly/server/session/session.go:303 +0x69
panic({0x966420?, 0xc000461df0?})
        /snap/go/10455/src/runtime/panic.go:914 +0x21f
github.com/df-mc/dragonfly/server/world.BlockRuntimeID({0xb62a10, 0xc0010814a0})
        /root/dragonfly/server/world/block.go:136 +0xd4
github.com/df-mc/dragonfly/server/session.stackFromItem({0x1873, {0xb58560, 0xc0010814a0}, 0x1, {0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, ...})
        /root/dragonfly/server/session/player.go:710 +0x1e7
github.com/df-mc/dragonfly/server/session.instanceFromItem({0x1873, {0xb58560, 0xc0010814a0}, 0x1, {0x0, 0x0}, {0x0, 0x0, 0x0}, 0x0, ...})
        /root/dragonfly/server/session/player.go:754 +0xbd
github.com/df-mc/dragonfly/server/session.(*Session).ViewEntity(0xc001112800, {0xb668c8, 0xc000722120})
        /root/dragonfly/server/session/world.go:128 +0x3e5
github.com/df-mc/dragonfly/server/world.showEntity({0xb668c8, 0xc000722120}, {0xb6d848, 0xc001112800})
        /root/dragonfly/server/world/world.go:1157 +0x37
github.com/df-mc/dragonfly/server/world.(*World).AddEntity(0xc0000c8540, {0xb668c8, 0xc000722120})
        /root/dragonfly/server/world/world.go:691 +0x348
github.com/df-mc/dragonfly/server/player.(*Player).Drop(0xc00077a000, {0x1872, {0xb58560, 0xc0010803f0}, 0x1, {0x0, 0x0}, {0x0, 0x0, 0x0}, ...})
        /root/dragonfly/server/player/player.go:2167 +0x399
github.com/df-mc/dragonfly/server/session.(*InventoryTransactionHandler).handleNormalTransaction(0xc00200fc40?, 0xc001112800?, 0xc001112800)
        /root/dragonfly/server/session/handler_inventory_transaction.go:106 +0x89b
github.com/df-mc/dragonfly/server/session.(*InventoryTransactionHandler).Handle(0x989800?, {0xb5e068?, 0xc00226f720}, 0xc001112800)
        /root/dragonfly/server/session/handler_inventory_transaction.go:24 +0x97
github.com/df-mc/dragonfly/server/session.(*Session).handlePacket(0xc001112800, {0xb5e068, 0xc00226f720})
        /root/dragonfly/server/session/session.go:432 +0x8d
github.com/df-mc/dragonfly/server/session.(*Session).handlePackets(0xc001112800)
        /root/dragonfly/server/session/session.go:312 +0xdc
created by github.com/df-mc/dragonfly/server/session.(*Session).Start in goroutine 1
        /root/dragonfly/server/session/session.go:217 +0xa5
exit status 2
```

DF Commit: 21eb18fd3299363f3dbb837edb2d8b91cf79e409

--- a/server/block/blast_furnace.go
+++ b/server/block/blast_furnace.go
@@ -20,13 +20,13 @@ type BlastFurnace struct {
 	*smelter
 
 	// Facing is the direction the blast furnace is facing.
-	Facing cube.Face
+	Facing cube.Direction
 	// Lit is true if the blast furnace is lit.
 	Lit bool
 }
 
 // NewBlastFurnace creates a new initialised blast furnace. The smelter is properly initialised.
-func NewBlastFurnace(face cube.Face) BlastFurnace {
+func NewBlastFurnace(face cube.Direction) BlastFurnace {
 	return BlastFurnace{
 		Facing:  face,
 		smelter: newSmelter(),
@@ -66,7 +66,7 @@ func (b BlastFurnace) UseOnBlock(pos cube.Pos, face cube.Face, _ mgl64.Vec3, w *
 		return false
 	}
 
-	place(w, pos, NewBlastFurnace(user.Rotation().Direction().Face().Opposite()), user, ctx)
+	place(w, pos, NewBlastFurnace(user.Rotation().Direction().Opposite()), user, ctx)
 	return placed(ctx)
 }
 
@@ -122,7 +122,7 @@ func (b BlastFurnace) DecodeNBT(data map[string]interface{}) interface{} {
 
 // allBlastFurnaces ...
 func allBlastFurnaces() (furnaces []world.Block) {
-	for _, face := range cube.HorizontalFaces() {
+	for _, face := range cube.Directions() {
 		furnaces = append(furnaces, BlastFurnace{Facing: face})
 		furnaces = append(furnaces, BlastFurnace{Facing: face, Lit: true})
 	}

--- a/server/block/furnace.go
+++ b/server/block/furnace.go
@@ -19,13 +19,13 @@ type Furnace struct {
 	*smelter
 
 	// Facing is the direction the furnace is facing.
-	Facing cube.Face
+	Facing cube.Direction
 	// Lit is true if the furnace is lit.
 	Lit bool
 }
 
 // NewFurnace creates a new initialised furnace. The smelter is properly initialised.
-func NewFurnace(face cube.Face) Furnace {
+func NewFurnace(face cube.Direction) Furnace {
 	return Furnace{
 		Facing:  face,
 		smelter: newSmelter(),
@@ -65,7 +65,7 @@ func (f Furnace) UseOnBlock(pos cube.Pos, face cube.Face, _ mgl64.Vec3, w *world
 		return false
 	}
 
-	place(w, pos, NewFurnace(user.Rotation().Direction().Face().Opposite()), user, ctx)
+	place(w, pos, NewFurnace(user.Rotation().Direction().Opposite()), user, ctx)
 	return placed(ctx)
 }
 
@@ -121,7 +121,7 @@ func (f Furnace) DecodeNBT(data map[string]interface{}) interface{} {
 
 // allFurnaces ...
 func allFurnaces() (furnaces []world.Block) {
-	for _, face := range cube.HorizontalFaces() {
+	for _, face := range cube.Directions() {
 		furnaces = append(furnaces, Furnace{Facing: face})
 		furnaces = append(furnaces, Furnace{Facing: face, Lit: true})
 	}

--- a/server/block/lectern.go
+++ b/server/block/lectern.go
@@ -20,7 +20,7 @@ type Lectern struct {
 	sourceWaterDisplacer
 
 	// Facing represents the direction the Lectern is facing.
-	Facing cube.Face
+	Facing cube.Direction
 	// Book is the book currently held by the Lectern.
 	Book item.Stack
 	// Page is the page the Lectern is currently on in the book.
@@ -57,7 +57,7 @@ func (l Lectern) UseOnBlock(pos cube.Pos, face cube.Face, _ mgl64.Vec3, w *world
 	if !used {
 		return false
 	}
-	l.Facing = user.Rotation().Direction().Face().Opposite()
+	l.Facing = user.Rotation().Direction().Opposite()
 	place(w, pos, l, user, ctx)
 	return placed(ctx)
 }
@@ -159,7 +159,7 @@ func (l Lectern) EncodeBlock() (string, map[string]any) {
 
 // allLecterns ...
 func allLecterns() (lecterns []world.Block) {
-	for _, f := range cube.HorizontalFaces() {
+	for _, f := range cube.Directions() {
 		lecterns = append(lecterns, Lectern{Facing: f})
 	}
 	return

--- a/server/block/smoker.go
+++ b/server/block/smoker.go
@@ -20,13 +20,13 @@ type Smoker struct {
 	*smelter
 
 	// Facing is the direction the smoker is facing.
-	Facing cube.Face
+	Facing cube.Direction
 	// Lit is true if the smoker is lit.
 	Lit bool
 }
 
 // NewSmoker creates a new initialised smoker. The smelter is properly initialised.
-func NewSmoker(face cube.Face) Smoker {
+func NewSmoker(face cube.Direction) Smoker {
 	return Smoker{
 		Facing:  face,
 		smelter: newSmelter(),
@@ -66,7 +66,7 @@ func (s Smoker) UseOnBlock(pos cube.Pos, face cube.Face, _ mgl64.Vec3, w *world.
 		return false
 	}
 
-	place(w, pos, NewSmoker(user.Rotation().Direction().Face().Opposite()), user, ctx)
+	place(w, pos, NewSmoker(user.Rotation().Direction().Opposite()), user, ctx)
 	return placed(ctx)
 }
 
@@ -122,7 +122,7 @@ func (s Smoker) DecodeNBT(data map[string]interface{}) interface{} {
 
 // allSmokers ...
 func allSmokers() (smokers []world.Block) {
-	for _, face := range cube.HorizontalFaces() {
+	for _, face := range cube.Directions() {
 		smokers = append(smokers, Smoker{Facing: face})
 		smokers = append(smokers, Smoker{Facing: face, Lit: true})
 	}

